source/NVDAObjects/UIA/__init__.py

```diff
@@ -255,28 +258,36 @@ def _getTextFromUIARange(self,range):
 		"""
 		return range.getText(-1)
 
-	def _getTextWithFields_text(self,textRange,formatConfig,UIAFormatUnits=UIAFormatUnits):
+	def _getTextWithFields_text(self,textRange,formatConfig,UIAFormatUnits=None):
 		"""
 		Yields format fields and text for the given UI Automation text range, split up by the first available UI Automation text unit that does not result in mixed attribute values.
 		@param textRange: the UI Automation text range to walk.
 		@type textRange: L{UIAHandler.IUIAutomationTextRange}
 		@param formatConfig: the types of formatting requested.
 		@ type formatConfig: a dictionary of NVDA document formatting configuration keys with values set to true for those types that should be fetched.
-		@param UIAFormatUnits: the UI Automation text units (in order of resolution) that should be used to split the text such that formatting won't have mixed values.
-		@type UIAFormatUnits: List of UI Automation Text Units 
+		@param UIAFormatUnits: the UI Automation text units (in order of resolution) that should be used to split the text such that formatting won't have mixed values. If None, then self.UIAFormatUnits will be used.
+		@type UIAFormatUnits: List of UI Automation Text Units or None
 		@rtype: a Generator yielding L{textInfos.FieldCommand} objects containing L{textInfos.FormatField} objects, and text strings.
 		"""
 		log.debug("_getTextWithFields_text start")
-		log.debug("Walking by UIA unit %s"%UIAFormatUnits[0])
-		for tempRange in iterUIARangeByUnit(textRange,UIAFormatUnits[0]):
+		if UIAFormatUnits:
+			unit=UIAFormatUnits[0]
+			furtherUIAFormatUnits=UIAFormatUnits[1:]
+		else:
+			unit=None

```

Comment:
Can we have a comment here saying that this will try the entire range first before any units?